### ISSUE TITLE
SSD1331 demo failing

### ISSUE THREAD

**Issue by sgjava** [lexus2k/ssd1306#136]
`sudo ./build_and_run.sh -p linux -f demos/ssd1331_demo 
make: Entering directory '/home/servadmin/ssd1306/examples'
make -C ../src -f Makefile.linux MCU= SDL_EMULATION= \
                     ADAFRUIT=
make[1]: Entering directory '/home/servadmin/ssd1306/src'
make[1]: Nothing to be done for 'default'.
make[1]: Leaving directory '/home/servadmin/ssd1306/src'
mkdir -p ../bld/demos/
cc -o ../bld/demos/ssd1331_demo.out -g -Os -w -ffreestanding -MD -g -Os -w -ffreestanding -I./demos/ssd1331_demo -I../src -Wall -Werror -Wl,--gc-sections -ffunction-sections -fdata-sections  ../bld/main.o ../bld/demos/ssd1331_demo/sova.o ../bld/demos/ssd1331_demo/ssd1331_demo.o -L../bld -lssd1306
../bld/demos/ssd1331_demo.out
Failed to initialize SPI: No such file or directory!
^Cmake: *** [Makefile.linux:52: flash] Interrupt
`
I assume there's a way to set SPI device and pis for DC and RESET? Also I shouldn't have to run as root as I have the permissions set for non -root user on SPI device.

**Comment by lexus2k** [comment]
Do you have spidev kernel module loaded?
Does the application have access privileges to kernel module driver?
What is the configuration of the target system?

**Comment by lexus2k** [comment]
Yes, it use spidev from userspace, but accessing to spi char devices requires root privileges

**Comment by sgjava** [comment]
I just use a script to set permissions https://github.com/sgjava/javauio/blob/main/scripts/uio-permissions.sh

**Comment by lexus2k** [comment]
Fixed: The default PROJECT path in Makefile.common was ssd1306_demo but the actual demos are under demos/ subdirectory. Fixed the default to demos/ssd1306_demo. To build the SSD1331 demo on Linux: cd examples && make -f Makefile.linux PROJECT=demos/ssd1331_demo
